### PR TITLE
MULE-12385: Fix: Some endpoints allow to define a reconnection strategy

### PR DESCRIPTION
**Pull request by facundovs**
In mule 2.x, it was allowed to define reconnection strategies within endpoints.
In mule 3.x, Reconnection strategies only must be defined in connectors or globally.
However, currently, in some connectors, such as HTTP, FTP, AMQP, Mail, so on, it's still possible to define reconnection strategies within endpoints (in the xml configuration, but it doesn't work).
A possible solution is to modify the scheme.
This change only be applied to 3.x.

### CODE REVIEW

File: distributions/standalone/src/main/resources/MIGRATION.txt [hunk 1]
```diff
@@ -16,6 +16,7 @@ MULE-10686: XML entity expansion in Jersey is now disabled by default because it
 MULE-10979: The default response timeout and default transaction timeout can't be configured using system properties on the command line or in the wrapper.conf file anymore. In replacement, use the configuration element. For example: <configuration defaultResponseTimeout="20000"  defaultTransactionTimeout="40000"/>.
 MULE-11118: The HTTP listener now replies with status code 503 when the thread pool is exhausted (and poolExhaustedAction="ABORT") instead of closing the socket.
 MULE-11825: In a DB template query, to set a DB param with the null value, you can use the "NULL" literal value. For example: <db:in-param name="name" defaultValue="NULL"/>
+MULE-12385: Reconnection Strategies can only be defined in connector components or globally (using <configuration> element). Defining reconnection strategies in endpoints isn't already supported.
```
Comment (pablokraan) [inline distributions/standalone/src/main/resources/MIGRATION.txt]: "Defining reconnection strategies in endpoints isn't already supported." I would rephrase this, to indicate that the XSD supported to configure it, but that it was not supported on runtime or that the configuration was not used

File: modules/spring-config/src/main/resources/META-INF/mule.xsd [hunk 2]
```diff
@@ -3670,7 +3670,6 @@
     <xsd:group name="defaultEndpointElements">
         <xsd:sequence>
             <xsd:group ref="commonDefaultEndpointElements"/>
-            <xsd:element ref="abstract-reconnection-strategy" minOccurs="0"/>
```
Comment (pablokraan) [inline modules/spring-config/src/main/resources/META-INF/mule.xsd]: Review this change with Studio guys

File: tests/integration/src/test/java/org/mule/test/reconnection/InvalidReconnectionStrategyTestCase.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+package org.mule.test.reconnection;
+
+import org.junit.Test;
+import org.mule.config.spring.AbstractSchemaValidationTestCase;
+import org.xml.sax.SAXException;
+
+public class InvalidReconnectionStrategyTestCase extends AbstractSchemaValidationTestCase
+{
+
+    @Test(expected = SAXException.class)
```
Comment (pablokraan) [inline tests/integration/src/test/java/org/mule/test/reconnection/InvalidReconnectionStrategyTestCase.java]: This exception is too generic. Can you check the error message?

File: modules/spring-config/src/main/resources/META-INF/mule.xsd (hunk 2 above)
Comment (facundovs) [inline modules/spring-config/src/main/resources/META-INF/mule.xsd]: I was talking to Pablo Cabrera and he asks me to create a JIRA to notify these changes.
